Dining/forms.py

```diff
@@ -97,10 +103,15 @@ def save(self):
         # also check if it has changed from earlier status
         old_list = DiningList.objects.get(id=self.instance.id)
 
-        if 0 < self.instance.dinner_cost_single != old_list.dinner_cost_single:
-            self.instance.dinner_cost_total = self.instance.dinner_cost_single * self.instance.diners
+        total_dinner_cost = self.cleaned_data['dinner_cost_total']
 
-        self.instance.save(update_fields=self.Meta.fields)
+        if total_dinner_cost > 0:
+            s_cost = total_dinner_cost / self.instance.diners.count()
+            # round up slightly, to remove missing cents
+            s_cost = round(s_cost+Decimal(0.0049), 2)
+            self.instance.dining_cost = s_cost
+
+        self.instance.save(update_fields=self.Meta.save_fields)
```

Comment:
`DiningPaymentForm.Meta` instead of `self.Meta` seems safer to me because maybe Django tampers with the instance `Meta`.

Reply:
I'll fix that